re-added summary of platforms to changes summary (#487)
cylc/cylc-doc: src/7-to-8/summary.rst

This supports :term:`graph branching` and it allows the scheduler to
correctly diagnose :term:`workflow completion`.

Platform Awareness
------------------

.. seealso::

   :ref:`Platforms at Cylc 8.<majorchangesplatforms>`

Cylc 7 was aware of individual job hosts - one selected a host using:
``[runtime][<namespace>][remote]host``.

Cylc 8 is aware of sets of host settings called

> :term:`[job] platforms <platform>`. To set a platform use

Reviewer: ```suggestion
:term:`[job] platforms <platform>`. To choose a platform for a task use
```

``[runtime][<namespace>]platform``

Hosts of a platform must share a file system and :term:`job runner`:
If one host is unavailable Cylc 8 can use other hosts
on the same platform to interact with task jobs.

> The same hosts can belong to different platforms, for example

Reviewer: ```suggestion
The same hosts can belong to multiple platforms, for example
```
(I think that's a bit clearer?)

> you might be able to use the same host to launch both background and slurm

Reviewer: ```suggestion
you might be able to use the same host to launch both background and Slurm
```

> you might be able to use your local computer to launch background or slurm
> jobs.

Reviewer: ```suggestion
you might be able to use the same host to launch both background and slurm
jobs.
```
("Your local computer" might raise unhelpful questions in user's minds, since their "local computer" is not likely to be  hooked up to their HPC in that way).

.. note::

   Cylc 8 will pick a sensible platform for your Cylc 7 settings,

>    These deprecated settings will be removed in future.

Reviewer: ```suggestion
   These deprecated settings will be removed in a future release.
```

>    These deprecated settings will be removed at Cylc 9.

Author: ???

```suggestion
   These deprecated settings will be removed in future.
```
